dashboard/app/controllers/pd/workshop_enrollment_controller.rb

```diff
@@ -212,6 +206,12 @@ def confirm_join_session
     redirect_to controller: 'pd/session_attendance', action: 'attend'
   end
 
+  private def build_redirect_for_non_teacher(page_name, workshop_id)
+    source_page = ERB::Util.url_encode('workshop enroll')
+    return_to = ERB::Util.url_encode("/pd/workshops/#{workshop_id}/enroll")
+    "/#{page_name}?source_page=#{source_page}&return_to=#{return_to}"
+  end
```

Comment:
I feel like I remember you leaving a cleanup comment on an old PR to combine this shared code into a private method, and I swore I did it but when I was looking at this I saw that I must've forgotten! So I'll include it here since it's in the same space.